File: SeleniumLab1/src/Lab1/tests/LoginTest.java
package Lab1.tests;

import org.junit.jupiter.api.*;
import static org.junit.jupiter.api.Assertions.*;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.support.ui.ExpectedConditions;

import Lab1.pages.LoginPage;
import Lab1.pages.InventoryPage;
import Lab1.utils.ReportManager;

import com.aventstack.extentreports.*;

import java.time.Duration;

public class LoginTest {

    WebDriver driver;
    WebDriverWait wait;
    LoginPage loginPage;

    static ExtentReports extent;
    ExtentTest test;

    @BeforeAll
    static void iniciarReporte() {
        extent = ReportManager.getReport();
    }

    @BeforeEach
    void setUp() {
        System.setProperty("webdriver.chrome.driver", "C:\\\\selenium webdriver\\\\ChromeDriver\\\\chromedriver-win64\\\\chromedriver.exe");

        driver = new ChromeDriver();
        wait = new WebDriverWait(driver, Duration.ofSeconds(5));

        driver.get("https://www.saucedemo.com/");
        driver.manage().window().maximize();

        loginPage = new LoginPage(driver);
    }

    @AfterEach
    void tearDown() {
        driver.quit();
    }

    @AfterAll
    static void cerrarReporte() {
        extent.flush(); // Genera el HTML
    }

    // TEST 1: Login correcto
    @Test
    void loginCorrecto() {

        test = extent.createTest("Login Correcto");

        loginPage.login("standard_user", "secret_sauce");

        wait.until(ExpectedConditions.urlContains("inventory"));

        if (driver.getCurrentUrl().contains("inventory")) {
            test.pass("Login exitoso");
        } else {
            test.fail("Login fallido");
        }

        assertTrue(driver.getCurrentUrl().contains("inventory"));
    }

    // TEST 2: Login incorrecto
    @Test
    void loginIncorrecto() {

        test = extent.createTest("Login Incorrecto");

        loginPage.login("usuario_incorrecto", "contraseña_incorrecta");

        wait.until(ExpectedConditions.visibilityOfElementLocated(
            By.cssSelector("[data-test='error']")
        ));

        if (loginPage.errorVisible()) {
            test.pass("Mensaje de error mostrado correctamente");
        } else {
            test.fail("No se mostró mensaje de error");
        }

        assertTrue(loginPage.errorVisible());
    }

    // TEST 3: Agregar producto al carrito
    @Test
    void agregarProductoAlCarrito() throws InterruptedException {

        test = extent.createTest("Agregar producto al carrito");

        loginPage.login("standard_user", "secret_sauce");

        wait.until(ExpectedConditions.urlContains("inventory"));

        InventoryPage inventory = new InventoryPage(driver);
        
        inventory.agregarBackpack();
        Thread.sleep(2000); // ver después del click

        wait.until(ExpectedConditions.visibilityOfElementLocated(
            By.cssSelector("[data-test='shopping-cart-badge']")
        ));

        int cantidad = inventory.obtenerCantidadCarrito();

        if (cantidad > 0) {
            test.pass("Producto agregado correctamente al carrito");
        } else {
            test.fail("No se agregó el producto al carrito");
        }

        assertTrue(inventory.carritoTieneProductos(),
                "El carrito debería mostrar productos");

        assertTrue(cantidad > 0,
                "El carrito debería tener al menos 1 producto");
    }
}

File: Lab1-python/LoginTest.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# Ruta al chromedriver
service = Service("C:\\selenium webdriver\\ChromeDriver\\chromedriver-win64\\chromedriver.exe")

driver = webdriver.Chrome(service=service)
wait = WebDriverWait(driver, 10)

try:
    # TEST 1: LOGIN INVÁLIDO
    driver.get("https://www.saucedemo.com/")
    driver.maximize_window()

    driver.find_element(By.ID, "user-name").send_keys("usuario_incorrecto")
    driver.find_element(By.ID, "password").send_keys("clave_incorrecta")
    driver.find_element(By.ID, "login-button").click()

    # Esperar mensaje de error
    error = wait.until(EC.visibility_of_element_located((By.CSS_SELECTOR, "[data-test='error']")))

    if error.is_displayed():
        print("Login inválido: OK (mensaje mostrado)")
    else:
        print("Login inválido: FALLÓ")

    # TEST 2: LOGIN CORRECTO + AGREGAR PRODUCTO
    driver.get("https://www.saucedemo.com/")

    driver.find_element(By.ID, "user-name").send_keys("standard_user")
    driver.find_element(By.ID, "password").send_keys("secret_sauce")
    driver.find_element(By.ID, "login-button").click()

    # Esperar inventario
    wait.until(EC.visibility_of_element_located((By.ID, "inventory_container")))

    # Esperar botón y hacer click
    wait.until(EC.element_to_be_clickable((By.ID, "add-to-cart-sauce-labs-backpack")))
    driver.find_element(By.ID, "add-to-cart-sauce-labs-backpack").click()
    print("Login correcto: OK")
    print("Producto agregado")

    # TEST 3: VALIDAR CARRITO
    badge = wait.until(EC.visibility_of_element_located(
        (By.CSS_SELECTOR, "[data-test='shopping-cart-badge']")
    ))

    cantidad = int(badge.text)

    if cantidad > 0:
        print("Carrito OK - cantidad:", cantidad)
    else:
        print("Carrito FALLÓ")

finally:
    driver.quit()